doc/aaa/ldap.md

```diff
@@ -0,0 +1,2617 @@
+     |           |  +--rw oc-aaa-ldap-ext:version?                      uint8
+     |           |  +--rw oc-aaa-ldap-ext:base?                         string
+     |           |  +--rw oc-aaa-ldap-ext:ssl?                          ldap-ssl-type
+     |           |  +--rw oc-aaa-ldap-ext:bind-dn?                      string
+     |           |  +--rw oc-aaa-ldap-ext:bind-pw?                      string
+     |           |  +--rw oc-aaa-ldap-ext:source-address?               oc-inet:ip-address
+     |           |  +--rw oc-aaa-ldap-ext:vrf-name?                     string
+     |           |  +--rw oc-aaa-ldap-ext:idle-time-limit?              uint32
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-group?               string
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-shadow?              string
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-netgroup?            string
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-sudoers?             string
+     |           |  +--rw oc-aaa-ldap-ext:nss-initgroups-ignoreusers?   string
+     |           |  +--rw oc-aaa-ldap-ext:pam-filter?                   string
+     |           |  +--rw oc-aaa-ldap-ext:pam-login-attribute?          string
+     |           |  +--rw oc-aaa-ldap-ext:pam-group-dn?                 string
+     |           |  +--rw oc-aaa-ldap-ext:pam-member-attribute?         string
+     |           |  +--rw oc-aaa-ldap-ext:sudoers-base?                 string
+     |           |  +--rw oc-aaa-ldap-ext:scope?                        enumeration
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-passwd?              string
+     |           +--ro oc-aaa-ldap-ext:state
+     |           +--rw oc-aaa-ldap-ext:maps
+     |              +--rw oc-aaa-ldap-ext:map* [name from]
+     |                 +--rw oc-aaa-ldap-ext:name      -> ../config/name
+     |                 +--rw oc-aaa-ldap-ext:from      -> ../config/from
+     |                 +--rw oc-aaa-ldap-ext:config
+     |                 |  +--rw oc-aaa-ldap-ext:name?   enumeration
+     |                 |  +--rw oc-aaa-ldap-ext:from?   string
+     |                 |  +--rw oc-aaa-ldap-ext:to?     string
+     |                 +--ro oc-aaa-ldap-ext:state
+     |                    +--ro oc-aaa-ldap-ext:name?   enumeration
+     |                    +--ro oc-aaa-ldap-ext:from?   string
+     |                    +--ro oc-aaa-ldap-ext:to?     string
+...
+...
+
+
+```
+
+## REST/gNMI
+
+REST, and gNMI support would be needed by writing transformer annotation,
+overloads, and SONiC Yang.
+
+## KLISH
+
+KLISH commands (XML, actioners, renderers) needs to be written for
+
+- AAA authentication ldap option.
+
+```
+sonic(config)# [no] aaa authentication login-method { ldap | local }*
+```
+
+- AAA authorization login ldap option..
+
+```
+sonic(config)# [no] aaa authorization login {ldap|local}
+```
+
+- AAA name-service.
+
+```
+sonic(config)# [no] aaa name-service passwd { ldap|login|local }
```

Comment:
To my knowledge no other vendor (including Cisco) provides a way to configure the Linux Name Service separately from AAA. Historically, vendors such as Cisco have deployed their own proprietary OSes (e.g. IOS), which don't require the equivalent of a Linux Name Service. I'm not familiar with Cisco's IOS, but I assume that it does not require that users be assigned a unique User ID (uid) or primary Group ID (gid) like Linux and UNIX require. 

If I'm not mistaking, Cisco also has a Linux-based OS and I'm wondering if they also allow configuring the Name Service separately from AAA or do they always make the Name Service match what has been configured for AAA. It'd be interesting to investigate.